config/internal/configsource/manager.go

```diff
@@ -305,12 +324,134 @@ func (m *Manager) expandConfigSources(ctx context.Context, s string) (interface{
 	return retrieved.Value(), nil
 }
 
+// expandString expands environment variables and config sources that are specified on the string.
+func (m *Manager) expandString(ctx context.Context, s string) (interface{}, error) {
```

Comment:
Would be nice to break down this func into smaller ones, it is a bit too long to read.

Reply:
Done (shaved about a dozen lines)